packages/sdks/src/functions/track/index.ts

```diff
@@ -118,7 +123,10 @@ export async function _track(eventProps: EventProps) {
     return;
   }
 
-  return fetch(`https://cdn.builder.io/api/v1/track`, {
+  const baseUrl = eventProps.apiHost || 'https://cdn.builder.io';
+
+  delete eventProps.apiHost;
```

Comment:
this approach is dangerous. it might delete `apiHost` from the parent function, which means subsequent uses of `apiHost` won't have access to the variable.

Here is an example:
![CleanShot 2024-10-28 at 11 49 30@2x](https://github.com/user-attachments/assets/45914203-84f5-472b-81be-1529b33c0d54)


so `parentFunction` now loses its `props.foo`.

Instead, what you should do is change:

```ts
export async function _track(eventProps: EventProps) {
```

to:

```ts
export async function _track(({ apiHost, ...eventProps }): EventProps) {
```

this does not involve deleting anything

Reply:
you still didn't get rid of the `delete` here. it's redundant and can be removed

Reply:
My bad, pushed the change